opendevin/controller/agent_controller.py

```diff
@@ -291,7 +294,16 @@ async def _step(self):
             logger.debug(f'[Agent Controller {self.id}] Delegate step done')
             assert self.delegate is not None
             delegate_state = self.delegate.get_agent_state()
-            if delegate_state == AgentState.ERROR:
+            logger.debug(
+                f'[Agent Controller {self.id}] Delegate state: {delegate_state}'
+            )
+            if delegate_state == AgentState.ERROR or (
+                self.headless_mode and delegate_state == AgentState.PAUSED
```

Comment:
I suggest we change `await self.set_agent_state_to(AgentState.PAUSED)` code to `await self.set_agent_state_to(AgentState.ERROR)` in headless mode [here](https://github.com/OpenDevin/OpenDevin/pull/2994/files#diff-37dff7c7cda815bf00598e78acaaebb91df0bf68b820bdca78a86b89f67dfc9bR358)

Reply:
Good idea! pushed the changes in the latest commits.